drop-analytics/src/lib.rs

```diff
@@ -43,14 +43,19 @@ pub struct FileInfo {
 }
 
 pub trait Moose: Send + Sync {
-    fn service_quality_initialization_init(&self, res: Result<(), i32>, phase: Phase);
-    fn service_quality_transfer_batch(&self, phase: Phase, transfer_id: String, info: TransferInfo);
+    fn service_quality_initialization_init(&self, res: Result<(), i32>);
+    fn service_quality_transfer_batch(
+        &self,
+        transfer_id: String,
+        info: TransferInfo,
+        protocol_version: i32,
+    );
     fn service_quality_transfer_file(
         &self,
         res: Result<(), i32>,
-        phase: Phase,
         transfer_id: String,
         transfer_time: i32,
+        direction: TransferDirection,
```

Comment:
Changing API must have been quite a pain. If I were you I would simply ignore the argument inside while leaving the API untouched.
But since you already did this job it's great :+1: